drivers/interrupt_controller/intc_gic.c

```diff
@@ -126,10 +137,15 @@ static void gic_dist_init(void)
 	sys_write32(0, GICD_CTLR);
 
 	/*
-	 * Set all global interrupts to all CPUs.
+	 * Enable all global interrupts distributing to CPUs listed
+	 * in dts with the count of CONFIG_MP_NUM_CPUS.
 	 */
+	for (i = 0; i < CONFIG_MP_NUM_CPUS; i++)
+		cpu_mask |= (1 << cpu_mpid_list[i]);
+	reg_val = cpu_mask | (cpu_mask << 8) | (cpu_mask << 16)
+		| (cpu_mask << 24);
```

Comment:
use `BIT_MASK`

Reply:
I don't think we can use BIT_MASK here, for example, if cpu_mask=0x28, then reg_val is expected to be 0x28282828. thanks.

Reply:
Oh, right.